### PR TITLE
[BUGFIX] Restore support for data- prefixed arguments

### PR DESCRIPTION
**Pull request by NamelessCoder**
Due to a regression in https://github.com/TYPO3/Fluid/pull/419,
a TagBasedViewHelper which was called with one or more
data- prefixed arguments would discard the prefixed arguments
and the output would not contain the attribute.

This patch fixes that regression by moving the handling of
unregistered arguments to the initialize() function in order
to make sure the additional attributes are handled *after*
the reset() method has been called on TagBuilder.

References: #419

### CODE REVIEW

**Comment by mbrodala** [comment]
@NamelessCoder Can you rebase this one? I guess #530 will need to be adjusted if merged after this one.

**Comment by NamelessCoder** [comment]
@mbrodala Done, but please wait for Travis to complete - I wasn't able to run unit tests locally and I'm a bit busy on other projects. Yep, #530 will need an adjustment afterwards!

File: tests/Functional/Cases/TagBasedTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        $invoker = new ViewHelperInvoker();
+        $viewHelper = new TagBasedTestViewHelper();
+        $arguments = [
+            'data-foo' => 'bar',
+        ];
+        $result = $invoker->invoke($viewHelper, $arguments, new RenderingContextFixture());
+        $this->assertSame('<div data-foo="bar" />', $result);
+    }
+}
```
Comment (mbrodala) [inline tests/Functional/Cases/TagBasedTest.php:48]: We should add one additional test here which passes in *both* `'data-foo'` and `'data' => ['foo' => ...]` to clearly document the expected result for such a conflict.

**Review by mbrodala — CHANGES_REQUESTED** [review]
Should I add this test?

**Comment by NamelessCoder** [comment]
> Should I add this test?

That would be great, thanks!

**Review by mbrodala — APPROVED** [review]

**Comment by mbrodala** [comment]
@NamelessCoder I'd merge this now if you don't see anything missing since it predates #530

**Comment by NamelessCoder** [comment]
@mbrodala All good from my perspective!

**Pull request merged**